File: go/189.go
package main

func rotate(nums []int, k int) {
  k %= len(nums)
	tmp, s := make([]int, len(nums)), len(nums)
	copy(tmp, nums)
	for i := 0; i < s; i++ {
		nums[k] = tmp[i]
		k = (k + 1) % s
	}
}

File: cpp/189.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  void rotate(vector<int> &nums, int k) {
    vector<int> tmp(nums.size(), 0);
    for (int i = 0; i < nums.size(); i++)
      tmp[i] = nums[(k + i) % nums.size()];
    nums = tmp;
  }
};
